heed/src/env.rs

```diff
@@ -260,6 +261,57 @@ pub enum CompactionOption {
 }
 
 impl Env {
+    /// Return the size used by all the databases in the environment.
+    pub fn size(&self) -> Result<usize> {
+        let mut size = 0;
+
+        let mut stat = std::mem::MaybeUninit::uninit();
+        unsafe { mdb_result(ffi::mdb_env_stat(self.0.env, stat.as_mut_ptr())) }?;
+        let stat = unsafe { stat.assume_init() };
+
+        size += (stat.ms_leaf_pages + stat.ms_branch_pages + stat.ms_overflow_pages)
+            * stat.ms_psize as usize;
+
+        let rtxn = self.read_txn()?;
+
+        let dbi = self.raw_open_dbi(&rtxn, None, 0)?;
+
+        // we don’t want anyone to open an environment while we’re computing the stats
+        // thus we take a lock on all mutexes
+        let locks = (
+            OPENED_ENV.write().unwrap(),
```

Comment:
I would prefer that you take a `read` lock here, as other threads that want to read the other envs should be able to do so. Am I right?

Reply:
In fact, it is not even required to take a lock on the `OPENED_ENV`, right? As the `Env` that is `&self` is an `Arc<EnvInner>`, the lock is therefore replicated on all the owners of this specific `Env`. You can simply take a lock on the `self.dbi_open_mutex`.